src/main/java/org/sagebionetworks/bridge/services/UploadService.java

```diff
@@ -527,6 +566,89 @@ public void uploadComplete(String appId, UploadCompletionClient completedBy, Upl
         // Save uploadedOn date and uploadId to related adherence records.
         updateAdherenceWithUploadInfo(appId, upload);
     }
+
+    public void redriveUpload( byte[] fileBytes) throws IOException {
+        // Convert the byte[] array to a String
+        String fileContent = new String(fileBytes);
+
+        // Split the content into lines (each line is an entry - uploadId)
+        String[] lines = fileContent.split("\\r?\\n");
+
+        // Count the number of entries (lines)
+        int numberOfEntries = lines.length;
+
+        // Redrive upload.
+        if (numberOfEntries <= 10) {
+            redriveSmallAmountOfUploads(fileBytes);
+        } else {
+            redriveLargeAmountOfUploads(fileBytes);
+        }
+    }
+
+    private void redriveSmallAmountOfUploads(byte[] fileBytes) throws JsonProcessingException {
+        // Convert the byte[] array to a String
+        String fileContent = new String(fileBytes);
+
+        // Split the String into an array of lines
+        String[] linesArray = fileContent.split("\n");
+
+        // Convert the array of lines into a List
+        List<String> uploadIds = new ArrayList<>(Arrays.asList(linesArray));
+
+        for (String uploadId : uploadIds) {
+            Upload upload = getUpload(uploadId);
+            String appId = upload.getAppId();
+            if (appId == null) {
+                appId = healthCodeDao.getAppId(upload.getHealthCode());
+            }
+            uploadComplete(appId, UploadCompletionClient.REDRIVE, upload, true);
+            UploadValidationStatus validationStatus = pollUploadValidationStatusUntilComplete(uploadId);
+            if (validationStatus.getStatus() != UploadStatus.SUCCEEDED) {
+                logger.error("Redrive failed for uploadId=" + uploadId + ": " + COMMA_SPACE_JOINER.join(
+                        validationStatus.getMessageList()));
+            }
+        }
+    }
+
+    private void redriveLargeAmountOfUploads(byte[] fileBytes) throws IOException {
+        // set up s3 Key.
+        String currentYear = String.valueOf(DateUtils.getCurrentDateTime().getYear());
+        String currentMonth = String.valueOf(DateUtils.getCurrentDateTime().getMonthOfYear());
+        String s3Key = REDRIVE_UPLOAD_S3_KEY_PREFIX + currentYear + "-" + currentMonth;
+
+        // Upload file to S3 bucket
+        InputStream inputStream = new ByteArrayInputStream(fileBytes);
+        ObjectMetadata metadata = new ObjectMetadata();
+        metadata.setContentType(URLConnection.guessContentTypeFromStream(inputStream));
+
+        PutObjectRequest request = new PutObjectRequest(CONFIG_KEY_BACKFILL_BUCKET, s3Key, inputStream, metadata);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: When objects are uploaded to Amazon S3 by using streams (either through an AmazonS3 client or TransferManager), you might encounter network connectivity or timeout issues.

**Fix**: The most reliable way to avoid a ResetException is to provide data by using a File or FileInputStream, which the AWS SDK for Java can handle without being constrained by mark and reset limits. If the input stream isn&#39;t a FileInputStream but does support mark and reset, you can set the mark limit by using the [setReadLimit](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/RequestClientOptions.html#setReadLimit-int-) method of [RequestClientOptions.](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/RequestClientOptions.html) Its default value is 128 KB. Setting the read limit value to one byte greater than the size of stream will reliably avoid a ResetException.

[Learn more](https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/best-practices.html)